Release/src/http/client/http_client_asio.cpp

```diff
@@ -917,6 +912,8 @@ class asio_context final : public request_context, public std::enable_shared_fro
             // 'start_http_request_flow'
             std::shared_ptr<ssl_proxy_tunnel> ssl_tunnel =
                 std::make_shared<ssl_proxy_tunnel>(shared_from_this(), start_http_request_flow);
+
+            m_timer.start_connecting();
```

Comment:
This replaces the above-mentioned call inside `ssl_proxy_tunnel::start_proxy_connect()`. All calls to `m_timer.start_connection()` are now consistently placed inside `asio_context::start_request()`.